### PR TITLE
feat: disable escaping for landing page content strings 

### LINKED ISSUE #4116: Disable automatic escaping for landing page content strings
**Is your feature request related to a problem? Please describe.**

At the moment it's not possible to style the content inside landing page macros. This is causing issues because it prevents content authors from using HTML entities (`&mdash;`) or basic Markdown styling (`*emphasis*`) inside these strings.

**Describe the solution you'd like**

It looks like the most direct solution is to update how strings inserted to process them as Markdown and avoid escaping HTML.

**site/_includes/macros/landing-deco.njk** starting on line 14
```diff
          <span>
-            {{ title }}
+            {{ title | md | safe }}
          </span>
        </h2>
        <p class="gap-top-300 gap-bottom-600 md-only">
-          {{ desc }}
+          {{ desc | md | safe }} 
        </p>
```

This change would also need to be applied to other landing macros that output similar strings.

**Describe alternatives you've considered**

The inability to use HTML entities is my primary concern. If the team would rather not support Markdown in these strings, we could just use the `safe` pipe instead of both `md` and `safe`.

The only other alternative I briefly considered is to use tags instead of macros, but I wrote this off as it would essentially be a complete rewrite of the landing page system.

**Additional context**

N/A

### PR DESCRIPTION
**Pull request by mamieorine**
Fixes #4116

Changes proposed in this pull request:

- Disable automatic escaping for landing page content strings

### CODE REVIEW

File: site/_includes/partials/toc-under-tree.njk [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{#
+  This is used as the TOC on smaller screens (<= 992) that makes use of the white space
```
Comment (devnook) [inline site/_includes/partials/toc-under-tree.njk:2]: Is this file intended for this Pr?

File: site/_includes/macros/landing-section.njk [hunk 2]
```diff
@@ -24,18 +24,18 @@
               <div class="item">
                 <div class="item__inner height-full display-flex direction-column">
                   <div class="item__title">
-                    <a href="{{ post.url }}">{{ post.title or post.data.title }}</a>
+                    <a href="{{ post.url }}">{{ post.title or post.data.title | md | safe }} </a>
```
Comment (devnook) [inline site/_includes/macros/landing-section.njk:27]: Once https://github.com/GoogleChrome/developer.chrome.com/pull/4431 lands, can we add examples of this component,  including markdown, to document & test that markdown is allowed and working?

Comment (mamieorine) [reply site/_includes/macros/landing-section.njk:27]: Yes, we can add the example of using markdown with this component into the 'add a landing page' page

File: site/_includes/partials/toc-under-tree.njk (hunk 1 above)
Comment (mamieorine) [reply site/_includes/partials/toc-under-tree.njk]: I have removed it

File: site/_includes/macros/landing-section.njk [hunk 3]
```diff
@@ -24,18 +24,18 @@
...
+                  <div class="desc flex-1">{{ post.description or post.data.description | md | safe }}</div>
                 </div>
               </div>
             </div>
           {% else %}
             <div class="item height-full">
               <div class="item__inner
                 height-full">
-                <div class="item__title"><a href="{{ post.url }}">{{ post.title or post.data.title }}</a></div>
-                <p class="desc flex-1">{{ post.description or post.data.description }}</p>
```
Comment (devnook) [inline site/_includes/macros/landing-section.njk:38]: Why change from p to div?

**Comment by devnook** [comment]
@mamieorine ptal

File: site/_includes/macros/landing-section.njk (hunk 3 above)
Comment (mamieorine) [reply site/_includes/macros/landing-section.njk:38]: I brought back the p tag. I accidentally changed it while updating this section

**Comment by mamieorine** [comment]
@devnook  I brought back the p tag and merged the main brach into this branch. Please help review this PR again. Thank you!

**Comment by devnook** [comment]
Can you also please add the example to the components page?

**Review by devnook — APPROVED** [review]
